### ISSUE TITLE
Native nodejs Request errors not supported ...

### ISSUE THREAD

**Issue by veljkopopovic** [moul/node-slumber#3]
within lib/API.js file, neither _Class.prototype.get, _Class.prototype["delete"], _Class.prototype.post,  _Class.prototype.put minds Nodejs request error argument in handler function ...
It's very simple to reproduce it, test against some site that does not exist (http://ggggagle.com) or malformed URL ... 
_this.wrap_response should be called if error exists, but I'm not sure how passed data will affect rest of library ... Will investigate a bit and create a PR later today ...

**Comment by moul** [comment]
Hey, thank you for the reporting

Can you just please send me a minimal script with the output of the script so I'm sure we talk about the same thing

**Comment by veljkopopovic** [comment]
Try to run something like this ...
```javascript
var slumber = require('slumber');
var api = slumber.API('http://somenonexistingsite.blah')('resourlce').get();
```

And you should get an output like
```
TypeError: Cannot read property 'statusCode' of undefined
    at Request._callback (/home/veljko/trash/slumber_fail/node_modules/slumber/lib/API.js:178:40)
    at self.callback (/home/veljko/trash/slumber_fail/node_modules/slumber/node_modules/request/request.js:122:22)
    at Request.EventEmitter.emit (events.js:95:17)
    at ClientRequest.self.clientErrorHandler (/home/veljko/trash/slumber_fail/node_modules/slumber/node_modules/request/request.js:231:10)
    at ClientRequest.EventEmitter.emit (events.js:95:17)
    at Socket.socketErrorListener (http.js:1547:9)
    at Socket.EventEmitter.emit (events.js:95:17)
    at net.js:830:16
    at process._tickCallback (node.js:415:13)

shell returned 8

```

**Comment by moul** [comment]
I just fixed it, can you give a try before I release a new version ?

**Comment by veljkopopovic** [comment]
Will do later tonight and let you know once I'm done :-D

**Comment by veljkopopovic** [comment]
Now I get
```
TypeError: Cannot read property 'length' of undefined
    at _Class.wrap_response (/home/veljko/trash/slumber_fail/node_modules/slumber/lib/API.js:158:17)
    at _Class.wrap_response (/home/veljko/trash/slumber_fail/node_modules/slumber/lib/API.js:4:59)
    at Request._callback (/home/veljko/trash/slumber_fail/node_modules/slumber/lib/API.js:179:26)
    at self.callback (/home/veljko/trash/slumber_fail/node_modules/slumber/node_modules/request/request.js:122:22)
    at Request.EventEmitter.emit (events.js:95:17)
    at ClientRequest.self.clientErrorHandler (/home/veljko/trash/slumber_fail/node_modules/slumber/node_modules/request/request.js:231:10)
    at ClientRequest.EventEmitter.emit (events.js:95:17)
    at Socket.socketErrorListener (http.js:1547:9)
    at Socket.EventEmitter.emit (events.js:95:17)
    at net.js:830:16
```
Note that I had no callback function in get as argument at all (script few comments above):
```JavaScript
var api = slumber.API('http://somenonexistingsite.blah')('resourlce').get();
```
If i do provide a cb, no output is given, no error or log about invalid URL ...

**Comment by moul** [comment]
Seems better, can you give a new try ?

(Thank you again :+1:)

**Comment by veljkopopovic** [comment]
Looks nice to me .... (Getting 'Missing callback error' ...)

Thx

**Comment by moul** [comment]
I just released a 0.6.0 version on npm with this fix and other ones from #4
